src/java.base/share/classes/jdk/internal/foreign/abi/s390x/sysv/TypeClass.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+/*
+ * ===========================================================================
+ * (c) Copyright IBM Corp. 2022, 2022 All Rights Reserved
+ * ===========================================================================
+ */
+
+package jdk.internal.foreign.abi.s390x.sysv;
+
+import java.lang.invoke.VarHandle;
+import java.util.List;
+
+import java.lang.foreign.GroupLayout;
+import java.lang.foreign.MemoryAddress;
+import java.lang.foreign.MemoryLayout;
+import java.lang.foreign.MemorySegment;
+import java.lang.foreign.ValueLayout;
+import static java.lang.foreign.ValueLayout.*;
+
+/**
+ * This class enumerates three argument types for Linux/s390x against the implementation
+ * of TypeClass on Linux/ppc64le.
+ */
+enum TypeClass {
+	INTEGER, /* Intended for all integral primitive types */
+	FLOAT,   /* Intended for float and double */
+	POINTER,
+	STRUCT,
+	STRUCT_ONE_FLOAT; /* Intended for a struct with only one float or double element */
+
+	static boolean isFloatingType(MemoryLayout layout) {
+		boolean isFloatOrDouble = false;
+
+		if ((layout instanceof ValueLayout)
```

Comment:
It is more clear to write it into 2 lines as:
```
((layout instanceof ValueLayout) && (classifyValueType((ValueLayout)layout) == FLOAT))
|| ((layout instanceof GroupLayout) && isStructWithOneFloat((GroupLayout)layout))
```

Reply:
Updated as suggested above.